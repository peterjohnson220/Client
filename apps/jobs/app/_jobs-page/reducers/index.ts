import { createFeatureSelector, createSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromJobsPageReducer from './jobs-page.reducer';
import * as fromModifyPricingsReducer from './modify-pricings.reducer';
import * as fromJobDescriptionReducer from './job-description.reducer';
import * as fromJobPeerMatchesReducer from './peer-matches-exchange.reducer';
import * as fromJobInsightsReducer from './job-insights.reducer';

// Feature area state
export interface JobsPageStateMain {
  jobsPage: fromJobsPageReducer.State;
  modifyPricings: fromModifyPricingsReducer.State;
  jobDescription: fromJobDescriptionReducer.State;
  peerMatches: fromJobPeerMatchesReducer.State;
  jobInsights: fromJobInsightsReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  jobsPage: JobsPageStateMain;
}

// Feature area reducers
export const reducers = {
  jobsPage: fromJobsPageReducer.reducer,
  modifyPricings: fromModifyPricingsReducer.reducer,
  jobDescription: fromJobDescriptionReducer.reducer,
  peerMatches: fromJobPeerMatchesReducer.reducer,
  jobInsights: fromJobInsightsReducer.reducer
};

// Select Feature Area
export const selectJobsPageMainState =
  createFeatureSelector<JobsPageStateMain>('jobsPageMain');

// Jobs Page Selectors
export const selectJobsPageState =
  createSelector(selectJobsPageMainState, (state: JobsPageStateMain) => state.jobsPage);

export const getJobsPageId = createSelector(selectJobsPageState, fromJobsPageReducer.getJobsPageId);
export const getCreatingProject = createSelector(selectJobsPageState, fromJobsPageReducer.getCreatingProject);
export const getChangingJobStatus = createSelector(selectJobsPageState, fromJobsPageReducer.getChangingJobStatus);
export const getDeletingJob = createSelector(selectJobsPageState, fromJobsPageReducer.getDeletingJob);
export const getStructureGradeNames = createSelector(selectJobsPageState, fromJobsPageReducer.getStructureGradeNames);
export const getCompanyPayMarkets = createSelector(selectJobsPageState, fromJobsPageReducer.getCompanyPayMarkets);
export const getPayMarketGroupedListItems = createSelector(selectJobsPageState, fromJobsPageReducer.getPayMarketGroupedListItems);
export const getExportOptions = createSelector(selectJobsPageState, fromJobsPageReducer.getExportOptions);
export const getNavigatingToOldPage = createSelector(selectJobsPageState, fromJobsPageReducer.getNavigatingToOldPage);
export const getExportEventId = createSelector(selectJobsPageState, fromJobsPageReducer.getExportEventId);
export const getExporting = createSelector(selectJobsPageState, fromJobsPageReducer.getExporting);

// Modify Pricings Selectors

export const selectModifyPricingsPageState =
  createSelector(selectJobsPageMainState, (state: JobsPageStateMain) => state.modifyPricings);

export const getDeletingPricing = createSelector(selectModifyPricingsPageState, fromModifyPricingsReducer.getDeletingPricing);
export const getUpdatingPricing = createSelector(selectModifyPricingsPageState, fromModifyPricingsReducer.getUpdatingPricing);
export const getDeletingPricingMatch = createSelector(selectModifyPricingsPageState, fromModifyPricingsReducer.getDeletingPricingMatch);
export const getUpdatingPricingMatch = createSelector(selectModifyPricingsPageState, fromModifyPricingsReducer.getUpdatingPricingMatch);
export const getRecalculatingPricingInfo = createSelector(selectModifyPricingsPageState, fromModifyPricingsReducer.getRecalculatingPricingInfo);

// Job Description Selectors
export const selectJobDescriptionState =
  createSelector(selectJobsPageMainState, (state: JobsPageStateMain) => state.jobDescription);

  export const getJobId = createSelector(selectJobDescriptionState, fromJobDescriptionReducer.getJobId);
  export const getLoading = createSelector(selectJobDescriptionState, fromJobDescriptionReducer.getLoading);
  export const getJobDescriptionSummary = createSelector(selectJobDescriptionState, fromJobDescriptionReducer.getJobDescriptionSummary);
  export const getUpdatedJobDescription = createSelector(selectJobDescriptionState, fromJobDescriptionReducer.getUpdatedJobDescription);

// Job Peer Matches
export const selectJobPeerMatchesState =
  createSelector(selectJobsPageMainState, (state: JobsPageStateMain) => state.peerMatches);

export const getPeerMatchesLoaded = createSelector(selectJobPeerMatchesState, fromJobPeerMatchesReducer.getPeerMatchesLoaded);
export const getPeerMatches = createSelector(selectJobPeerMatchesState, fromJobPeerMatchesReducer.getPeerMatches);

// Job Insights
export const selectJobInsightsState =
  createSelector(selectJobsPageMainState, (state: JobsPageStateMain) => state.jobInsights);

export const getJobInsights = createSelector(selectJobInsightsState, fromJobInsightsReducer.getJobInsights);
export const getJobCustomFields = createSelector(selectJobInsightsState, fromJobInsightsReducer.getJobCustomFields);
