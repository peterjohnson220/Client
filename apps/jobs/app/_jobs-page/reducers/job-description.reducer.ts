import cloneDeep from 'lodash/cloneDeep';

import { JobDescriptionSummary, AsyncStateObj, generateDefaultAsyncStateObj } from 'libs/models';
import { AsyncStateObjHelper } from 'libs/core';

import * as fromJobDescriptionActions from '../actions';

export interface State {
  jobId: number;
  updatedJobDescription: string;
  jobDescriptionSummary: AsyncStateObj<JobDescriptionSummary>;
}

export const initialState: State = {
  jobId: null,
  updatedJobDescription: null,
  jobDescriptionSummary: generateDefaultAsyncStateObj<JobDescriptionSummary>(null),
};

export function reducer(state = initialState, action: fromJobDescriptionActions.JobDescriptionActions): State {
  switch (action.type) {
    case fromJobDescriptionActions.LOAD_JOB_DESCRIPTION: {
      return {
        ...AsyncStateObjHelper.loading(state, 'jobDescriptionSummary'),
        jobId: action.payload
      };
    }
    case fromJobDescriptionActions.LOAD_JOB_DESCRIPTION_SUCCESS: {
      return AsyncStateObjHelper.loadingSuccess(state, 'jobDescriptionSummary', action.payload);
    }
    case fromJobDescriptionActions.CHANGE_JOB_DESCRIPTION: {
      return {
        ...state,
        updatedJobDescription: action.payload,
        jobDescriptionSummary: {
          ...state.jobDescriptionSummary,
          savingSuccess: false,
          loadingError: false,
          savingError: false
        }
      };
    }
    case fromJobDescriptionActions.SAVE_JOB_DESCRIPTION: {
      return AsyncStateObjHelper.saving(state, 'jobDescriptionSummary');
    }
    case fromJobDescriptionActions.SAVE_JOB_DESCRIPTION_SUCCESS: {
      const jobSummaryClone: AsyncStateObj<JobDescriptionSummary> = cloneDeep(state.jobDescriptionSummary);
      jobSummaryClone.obj.JobSummary = action.payload;
      jobSummaryClone.saving = false;
      jobSummaryClone.savingSuccess = true;
      return {
        ...state,
        jobDescriptionSummary: jobSummaryClone
      };
    }
    case fromJobDescriptionActions.SAVE_JOB_DESCRIPTION_ERROR: {
      return AsyncStateObjHelper.savingError(state, 'jobDescriptionSummary');
    }
    default: {
      return state;
    }
  }
}

export const getJobId = (state: State) => state.jobId;
export const getLoading = (state: State) => {
  return state.jobDescriptionSummary.loading || state.jobDescriptionSummary.saving;
};
export const getJobDescriptionSummary = (state: State) => state.jobDescriptionSummary;
export const getUpdatedJobDescription = (state: State) => state.updatedJobDescription;


