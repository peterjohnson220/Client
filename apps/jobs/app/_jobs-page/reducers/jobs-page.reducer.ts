import * as fromJobsPageActions from '../actions';

export interface State {
  company: string;
  jobsPageId: string;
  loading: boolean;
  loadingError: boolean;
  addingToProject: boolean;
  showAddToProjectSummaryModal: boolean;
  pricingIdToBeDeleted: number;
}

export const initialState: State = {
  company: '',
  loading: false,
  jobsPageId: '',
  loadingError: false,
  addingToProject : false,
  showAddToProjectSummaryModal: false,
  pricingIdToBeDeleted: undefined,
};

export function reducer(state = initialState, action: fromJobsPageActions.JobsPageActions): State {
  switch (action.type) {
    case fromJobsPageActions.SET_JOBS_PAGE_ID: {
      return {
        ...state,
        jobsPageId: action.payload,
      };
    }
    case fromJobsPageActions.LOAD_COMPANY: {
      return {
        ...state,
        loading: true,
        loadingError: false,
      };
    }
    case fromJobsPageActions.LOAD_COMPANY_SUCCESS: {
      return {
        ...state,
        company: action.payload,
        loading: false,
        loadingError: false,
      };
    }
    case fromJobsPageActions.ADD_TO_PROJECT_SUMMARY: {
      return {
        ...state,
        showAddToProjectSummaryModal: true

      };
    }
    case fromJobsPageActions.CANCEL_ADD_TO_PROJECT_SUMMARY: {
      return {
        ...state,
        showAddToProjectSummaryModal: false
      };
    }
    case fromJobsPageActions.ADD_TO_PROJECT: {
      return {
        ...state,
        addingToProject: true
      };
    }
    case fromJobsPageActions.HANDLE_API_ERROR: {
      return {
        ...state,
        loading: false,
        loadingError: true
      };
    }
    case fromJobsPageActions.CONFIRM_DELETE_PRICING_FROM_GRID: {
      return {
        ...state,
        pricingIdToBeDeleted: action.payload.CompanyJobPricingId
      };
    }
    case fromJobsPageActions.CANCEL_DELETE_PRICING:
    case fromJobsPageActions.DELETE_PRICING_SUCCESS: {
      return {
        ...state,
        pricingIdToBeDeleted: undefined
      };
    }
    default: {
      return state;
    }
  }
}

export const getCompany = (state: State) => state.company;
export const getJobsPageId = (state: State) => state.jobsPageId;
export const getloading = (state: State) => state.loading;
export const getloadingError = (state: State) => state.loadingError;
export const getAddToProjectButtonState = (state: State) => state.addingToProject;
export const getPricingIdToBeDeleted = (state: State) => state.pricingIdToBeDeleted;
export const getShowAddToProjectSummaryModal = (state: State) => state.showAddToProjectSummaryModal;
