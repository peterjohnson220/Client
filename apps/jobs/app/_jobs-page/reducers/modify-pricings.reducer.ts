import { MatchedSurveyJob } from 'libs/models/payfactors-api/survey-search/response';

import * as fromModifyPricingsActions from '../actions';

export interface State {
  pricingsToModify: MatchedSurveyJob[];
  isModifyingPricings: boolean;
}

export const initialState: State = {
  pricingsToModify: undefined,
  isModifyingPricings: false
};

export function reducer(state = initialState, action: fromModifyPricingsActions.ModifyPricingsActions): State {
  switch (action.type) {
    case fromModifyPricingsActions.GET_PRICINGS_TO_MODIFY_SUCCESS:
      return {
        ...state,
        isModifyingPricings: true,
        pricingsToModify: action.payload
      };
    case fromModifyPricingsActions.MODIFY_PRICINGS_CANCEL:
      return {
        ...state,
        isModifyingPricings: false,
        pricingsToModify: undefined
      };
    default:
      return state;
  }
}

export const getPricingsToModify = (state: State) => state.pricingsToModify;
export const getIsModifyingPricings = (state: State) => state.isModifyingPricings;
