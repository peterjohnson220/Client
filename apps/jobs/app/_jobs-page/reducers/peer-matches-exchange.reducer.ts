import * as fromJobPeerMatchesActions from '../actions';

export interface State {
  jobId: number;
  peerMatchesLoaded: boolean;
  peerMatches: [];
}

export const initialState: State = {
  jobId: null,
  peerMatchesLoaded: false,
  peerMatches: undefined

};

export function reducer(state = initialState, action: fromJobPeerMatchesActions.JobPeerMatchesActions): State {
  switch (action.type) {
    case fromJobPeerMatchesActions.LOAD_JOB_PEER_MATCHES_SUCCESS: {
      return {
        ...state,
        peerMatchesLoaded: true,
        peerMatches: action.payload
      };
    }
    case fromJobPeerMatchesActions.LOAD_JOB_PEER_MATCHES: {
      return {
        ...state,
        peerMatchesLoaded: false
      };
    }
    default: {
      return state;
    }
  }
}
export const getPeerMatches = (state: State) => state.peerMatches;
export const getPeerMatchesLoaded = (state: State) => state.peerMatchesLoaded;





