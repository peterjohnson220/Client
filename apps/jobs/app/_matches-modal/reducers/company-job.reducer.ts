// Import all exports from our feature's actions
import * as fromCompanyJobActions from '../actions/company-job.actions';
import { CompanyJob } from 'libs/models/company/company-job.model';

// Define our feature state
export interface State {
  loading: boolean;
  loadingError: boolean;
  companyJob: CompanyJob;
}

// Define our initial state
const initialState: State = ({
  loading: false,
  loadingError: false,
  companyJob: null
});

// Reducer function
export function reducer(
  state = initialState,
  action: fromCompanyJobActions.Actions
): State {
  switch (action.type) {
    case fromCompanyJobActions.LOADING: {
      return {
        ...state,
        loading: true,
        loadingError: false
      };
    }
    case fromCompanyJobActions.LOADING_SUCCESS: {
      return {
        ...state,
        companyJob: action.payload[0],
        loading: false,
        loadingError: false
      };
    }
    case fromCompanyJobActions.LOADING_ERROR: {
      return {
        ...state,
        loading: false,
        loadingError: true
      };
    }
    default: {
      return state;
    }
  }
}

// Selector functions
export const getLoading = (state: State) => state.loading;
export const getLoadingError = (state: State) => state.loadingError;
export const getCompanyJob = (state: State) => state.companyJob;
