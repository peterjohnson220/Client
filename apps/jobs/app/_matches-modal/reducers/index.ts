import { createSelector, createFeatureSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromCompanyJobReducer from './company-job.reducer';
import * as fromMatchesReducer from './matches.reducer';


// Feature area state
export interface MatchesModalState {
  companyJob: fromCompanyJobReducer.State;
  matches: fromMatchesReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  matchesModal: MatchesModalState;
}

// Feature area reducers
export const reducers = {
  companyJob: fromCompanyJobReducer.reducer,
  matches: fromMatchesReducer.reducer
};

// Select Feature Area
export const selectMatchesModalState = createFeatureSelector<MatchesModalState>('matchesModalArea');

// Feature Selectors
export const selectCompanyJobState = createSelector(selectMatchesModalState, (state: MatchesModalState) => state.companyJob);
export const selectMatchesState = createSelector(selectMatchesModalState, (state: MatchesModalState) => state.matches);


export const getCompanyJob = createSelector(selectCompanyJobState, fromCompanyJobReducer.getCompanyJob);
export const getCompanyJobLoading = createSelector(selectCompanyJobState, fromCompanyJobReducer.getLoading);
export const getCompanyJobLoadingError = createSelector(selectCompanyJobState, fromCompanyJobReducer.getLoadingError);

export const {
  selectAll: getMatches
} = fromMatchesReducer.adapter.getSelectors(selectMatchesState);
export const getMatchesLoading = createSelector(selectMatchesState, fromMatchesReducer.getLoading);
export const getMatchesLoadingError = createSelector(selectMatchesState, fromMatchesReducer.getLoadingError);
