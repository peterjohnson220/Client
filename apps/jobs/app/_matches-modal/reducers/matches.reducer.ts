// Import all exports from our feature's actions
import { createEntityAdapter, EntityAdapter, EntityState } from '@ngrx/entity';

import { Match } from 'libs/models/company';

import * as fromMatchesActions from '../actions/matches.actions';


// Define our feature state
export interface State extends EntityState<Match> {
  loading: boolean;
  loadingError: boolean;
}

export const adapter: EntityAdapter<Match> = createEntityAdapter<Match>({
  selectId: (match: Match) => match.JobId
});

// Define our initial state
const initialState: State = adapter.getInitialState({
  loading: false,
  loadingError: false
});

// Reducer function
export function reducer(
  state = initialState,
  action: fromMatchesActions.Actions
): State {
  switch (action.type) {
    case fromMatchesActions.LOADING: {
      return {
        ...state,
        loading: true,
        loadingError: false
      };
    }
    case fromMatchesActions.LOADING_SUCCESS: {
      return {
        ...adapter.setAll(action.payload, state),
        loading: false,
        loadingError: false
      };
    }
    case fromMatchesActions.LOADING_ERROR: {
      return {
        ...state,
        loading: false,
        loadingError: true
      };
    }
    case fromMatchesActions.UPDATE_EXCLUDE_FROM_PARTICIPATION: {
      return {
        ...state
      };
    }
    default: {
      return state;
    }
  }
}

// Selector functions
export const getLoading = (state: State) => state.loading;
export const getLoadingError = (state: State) => state.loadingError;
