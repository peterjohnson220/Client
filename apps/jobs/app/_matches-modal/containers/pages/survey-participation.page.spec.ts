import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
import { ActivatedRoute } from '@angular/router';

import { combineReducers, Store, StoreModule } from '@ngrx/store';
import { DragulaModule } from 'ng2-dragula';

import * as fromRootState from 'libs/state/state';

import { SurveyParticipationPageComponent } from '.';
import * as fromReducers from '../../reducers';


describe( 'SurveyParticipationPageComponent', () => {
  let fixture: ComponentFixture<SurveyParticipationPageComponent>;
  let instance: SurveyParticipationPageComponent;
  let store: Store<fromReducers.State>;
  const queryStringParams = { 'companyJobId': 1 };

  beforeEach(waitForAsync( () => {
    TestBed.configureTestingModule( {
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          surveyParticipationReducers: combineReducers(fromReducers.reducers)
        }),
        DragulaModule.forRoot()
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: {
            snapshot: { queryParams: { keys: queryStringParams } }
          }
        }
      ],
      declarations: [
        SurveyParticipationPageComponent
      ],
      schemas: [NO_ERRORS_SCHEMA]
    });

    store = TestBed.inject(Store);
    fixture = TestBed.createComponent(SurveyParticipationPageComponent);
    instance = fixture.componentInstance;
  }));

  it( 'should show survey participation modal page', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
