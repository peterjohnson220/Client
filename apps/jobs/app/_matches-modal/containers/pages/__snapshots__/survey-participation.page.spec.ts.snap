// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyParticipationPageComponent should show survey participation modal page 1`] = `
<pf-survey-participation-page
  companyJob$={[Function Store]}
  companyJobLoading$={[Function Store]}
  dragEnd$={[Function AnonymousSubject]}
  dragMatches={[Function Subscription]}
  dragStart$={[Function AnonymousSubject]}
  dragulaService={[Function DragulaService]}
  isDragging="false"
  isDragging$={[Function Observable]}
  matches$={[Function Store]}
  matchesLoading$={[Function Store]}
  matchesModalAreaStore={[Function Store]}
  route={[Function Object]}
>
  <pf-async-container>
    
    <div
      class="matches-modal-container"
    >
      <div
        class="matches-modal-content matches-modal-title"
      >
        
        
      </div>
      <div
        class="matches-modal-content"
      >
         Use the survey participation function to determine which of your matches you would like to include in Survey Participation reports. Drag and drop your matches to the corresponding category to include or exclude them from participation. 
      </div>
      <div
        class="matches-modal-content font-weight-bold"
      >
         Included in Participation - These matches will be included in your survey participation report 
      </div>
      <div
        class="matches-modal-content matches-list-container dropzone"
        dragula="matches-bag"
        id="includeInParticipation"
        ng-reflect-dragula="matches-bag"
      >
        
        
      </div>
      <div
        class="matches-modal-content font-weight-bold"
      >
         Excluded from Participation - These matches will be excluded from your survey participation report 
      </div>
      <div
        class="matches-modal-content matches-list-container dropzone"
        dragula="matches-bag"
        id="excludeFromParticipation"
        ng-reflect-dragula="matches-bag"
      >
        
        
      </div>
    </div>
  </pf-async-container>
</pf-survey-participation-page>
`;
